subprojects/problems/src/main/java/org/gradle/api/problems/internal/DefaultProblemCategory.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.problems.internal;
+
+import org.gradle.api.problems.ProblemCategory;
+import org.gradle.util.Path;
+
+import java.util.List;
+
+import static java.lang.String.join;
+
+/**
+ * Represents a category of problems.
+ */
+public class DefaultProblemCategory implements ProblemCategory {
+
+    public static final int NAMESPACE_START_INDEX = 0;
+    public static final String GRADLE_PLUGIN_MARKER = "gradle-plugin";
+    public static final String DEPRECATION = "deprecation";
+    public static final String VALIDATION = "validation";
```

Comment:
ad naming:
the marker string is something that qualifies the fact, that it is coming from a plugin. I think that justifies the different naming compared to the categories.

ad enum:
I think it hides the fact that we are handling it as string in most cases and non predefined categories can't be added to a enum. This might lead to internal and external users thinking they are restricted to the enum value.

Reply:
On the enum side, as I said, I would see a predefined+custom mix.

Basically the user, when writes the code, would see that two methods are available: one with an enum (could have a descriptive name, like `PredefinedProblemCategories`), and a `String` for custom values. The JavaDoc could also explain this situation.

I'm just afraid of category fragmentation, as the users need to discover the psfs-s, and this is not trivial (i.e. Android). I believe having a type-system supported, discoverable, expandable/retractable set of defaults is beneficial for both sides.

I'm not willing to die on this hill though, as this is also a fine approach.